moto/dynamodb2/models.py

```diff
@@ -110,27 +111,27 @@ def describe_attrs(self, attributes):
         }
 
     def update(self, update_expression, expression_attribute_names, expression_attribute_values):
-        ACTION_VALUES = ['SET', 'set', 'REMOVE', 'remove']
-
-        action = None
-        for value in update_expression.split():
-            if value in ACTION_VALUES:
-                # An action
-                action = value
-                continue
-            else:
+        parts = [p for p in re.split(r'\b(SET|REMOVE|ADD|DELETE)\b', update_expression) if p]
```

Comment:
Why are you filtering values in the list comprehension with `if p`? As far as I understand a valid expression will not yield e.g. an empty string.